### PR TITLE
feat(app): add sidebar Menu component

### LINKED ISSUE #11: Create MenuItem component
## Figma Reference
https://www.figma.com/file/uEckQY4EaQ5rXwyISg6d1g/Fuel-Wallet?node-id=34%3A4757

## Props
- `icon: Icon`
- `label: string`

## Observations
- When this component is a children another `MenuItem`, it will have a file-three like indicating that is a submenu
- When this component has a children, need to have a toggle feature

## Implementation
```jsx
<MenuItem icon="Wallet" label="Wallet">
  <MenuItem icon="User" label="MyAccount" />
</MenuItem>
```

### PR DESCRIPTION
**Pull request by pedronauck**
✨ Fix #11

### CODE REVIEW

File: packages/app/src/systems/Sidebar/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+export type MenuProps = {
+  items: MenuItemObj[];
+};
+
+export function Menu({ items }: MenuProps) {
+  const [opened, setOpened] = useState<string | null>(null);
+  const navigate = useNavigate();
+
+  function handleAction(key: string | number) {
```
Comment (LuizAsFight) [inline packages/app/src/systems/Sidebar/components/Menu/Menu.tsx:83]: we can remove the tslint comment in beggining of this file.

changing this function to:


```
function handleAction(key: string | number) {
    const item = items.find((item) => item.key === key);
    if (item?.submenu) {
      setOpened(opened !== item.key ? item.key : null);
    } else if (item?.path) {
      navigate(item.path);
    }
  }
```

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
minor change requested.

paused reviewing because build is broken and can't test more.
will review again when it gets fixed

File: packages/app/src/systems/Sidebar/components/Menu/Menu.tsx (hunk 1 above)
Comment (pedronauck) [reply packages/app/src/systems/Sidebar/components/Menu/Menu.tsx:82]: ✅ @LuizAsFight

**Review by luizstacio — CHANGES_REQUESTED** [review]
When clicking on an item of the submenu, it the close animation, looks a bit buggy.

**Review by luizstacio — APPROVED** [review]

**Review by LuizAsFight — APPROVED** [review]

**Pull request merged**